### PR TITLE
Introduce metadata struct & compatibility

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

File: src/db.rs [hunk 1]
```diff
@@ -96,6 +96,7 @@ impl std::hash::Hasher for IdentityKeyHash {
 struct DbInner {
 	columns: Vec<Column>,
 	options: Options,
+	metadata: Metadata,
```
Comment (cheme) [inline src/db.rs:99]: ColumnOptions being stored in both metadata and options, does not look problematic.

File: src/options.rs [hunk 2]
```diff
@@ -23,7 +23,7 @@ use crate::column::Salt;
 use crate::compress::CompressionType;
 use rand::Rng;
 
-const CURRENT_VERSION: u32 = 4;
+pub const CURRENT_VERSION: u32 = 4;
 const LAST_SUPPORTED_VERSION: u32 = 3;
```
Comment (cheme) [inline src/options.rs:27]:
```suggestion
// TODO on last supported 4, remove `ValueTable` `no_compression` field.
const LAST_SUPPORTED_VERSION: u32 = 3;
```
or similar comment to not forget it.

**Review by cheme — APPROVED** [review]
LGTM

File: src/db.rs (hunk 1 above)
Comment (arkpar) [reply src/db.rs:99]: Semantically columns settings in options have a meaning of what the database metadata should be when it is created anew. Might be a good idea to split `Db::open` into `create` that takes column settings and `open` that does not.

**Pull request merged**